protocols/common.go

```diff
@@ -50,6 +51,7 @@ type DirectFundingObjectiveState struct {
 	ChannelId types.Bytes32
 
 	ParticipantIndex map[types.Address]uint // the index for each participant
+	ExpectedStates   []state.State          // indexed by turn number
```

Comment:
💡 it's worth considering storing a list of expected hashes (which are computed during initialization)